tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Program.cs

```diff
@@ -414,50 +411,165 @@ private static async Task Run(RunOptions options)
             }
         }
 
-        private static async Task WriteResults(string outputJson, string outputCsv, List<Result> results)
+        private static async Task WriteResults(string outputJson, string outputCsv, string outputTxt, string outputMd, List<Result> results)
         {
             using (var stream = File.OpenWrite(outputJson))
             {
                 await JsonSerializer.SerializeAsync(stream, results, JsonOptions);
             }
 
+            using (var streamWriter = new StreamWriter(outputCsv))
+            {
+                await WriteResultsSummary(streamWriter, results, OutputFormat.Csv);
+            }
+
+            using (var streamWriter = new StreamWriter(outputTxt))
+            {
+                await WriteResultsSummary(streamWriter, results, OutputFormat.Txt);
+            }
+
+            using (var streamWriter = new StreamWriter(outputMd))
+            {
+                await WriteResultsSummary(streamWriter, results, OutputFormat.Md);
+            }
+        }
+
+        public static async Task WriteResultsSummary(StreamWriter streamWriter, IEnumerable<Result> results, OutputFormat outputFormat)
+        {
             var groups = results.GroupBy(r => (r.Language, r.LanguageVersion, r.Service, r.Test, r.Arguments));
+
             var resultSummaries = groups.Select(g =>
             {
+                var requestedPackageVersions = g.Select(r => r.PackageVersions).Distinct(new DictionaryEqualityComparer<string, string>());
+
+                var runtimePackageVersions = requestedPackageVersions.Select(req =>
+                    g.Where(r => r.PackageVersions == req).First().Iterations.FirstOrDefault()?.PackageVersions);
```

Comment:
should this call language specific build tool to print out and extract actual versions from there?

Reply:
It's already doing this indirectly.  When the tests are run, `IterationResult.PackageVersions` should be populated with the runtime versions, and this code retrieves the versions and prints them.  It's just not implemented for Java yet (#3531).  Here's the implementation for .NET:

https://github.com/Azure/azure-sdk-tools/blob/3103c200ee4da1bc5dfa88988618b51980939570/tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Net.cs#L103-L116